src/main/ruby/core/truffle/cext.rb

```diff
@@ -124,9 +124,20 @@ def data_holder
     end
   end
 
+  ENCODING_CACHE = {}
+  ENCODING_CACHE_MUTEX = Mutex.new
+
   class RbEncoding
+    private_class_method :new
```

Comment:
It's a bit odd to me to have this above constants.
Also, the ENCODING_ could be here too, under RbEncoding.

Reply:
I like having declarations like that as the first thing you see in the class, rather than buried in the  middle. But I'll change it.

Reply:
It's highly related to the caching here, so it seems good to have them next to each other.